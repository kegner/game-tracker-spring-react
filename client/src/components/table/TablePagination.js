import React from 'react'

const TablePagination = (props) => {

  const { pageIndex, setPageIndex, gamesPages } = props;

  const pageButtonHandler = (index) => {
    setPageIndex(index);
  }

  return (
    <nav aria-label="Page navigation example">
      <ul className="pagination my-1">
        <li className="page-item">
          <button className="page-link"
            onClick={() => pageButtonHandler(pageIndex - 1)}
            disabled={pageIndex === 0}
          >
            Previous
          </button>
        </li>
        {gamesPages.map((_, index) => {
          const activeClass = index === pageIndex ? " active" : "";
          return <li key={index} className={`page-item ${activeClass}`}>
            <button className="page-link"
              onClick={() => pageButtonHandler(index)}
            >
              {index + 1}
            </button>
          </li>
        })}
        <li className="page-item">
          <button className="page-link"
            onClick={() => pageButtonHandler(pageIndex + 1)}
            disabled={pageIndex === gamesPages.length - 1}
          >
            Next
          </button>
        </li>
      </ul>
    </nav>
  )
}

export default TablePagination
