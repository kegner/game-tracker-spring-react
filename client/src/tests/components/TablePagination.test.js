import React from 'react';
import { Provider } from "react-redux";
import { render, fireEvent, getByText } from '@testing-library/react';
import { createTestStore } from '../store/testStore';
import TablePagination from '../../components/table/TablePagination';

let hasClicked = false;
let indexNumber = -1;
const clickHandler = (index) => {
  hasClicked = true;
  indexNumber = index;
}

let store;
let dispatch;
let component;
let container;
let asFragment;
let gamesPages = [
  [
    { id: "1", title: "Game 1" },
    { id: "2", title: "Game 2" },
    { id: "3", title: "Game 3" },
    { id: "4", title: "Game 4" },
    { id: "5", title: "Game 5" },
  ],
  [
    { id: "6", title: "Game 6"}
  ]
];

describe("TablePagination", () => {

  beforeEach(() => {
    hasClicked = false;
    indexNumber = -1;
    store = createTestStore({
      mockStore: true
    });
    dispatch = store.dispatch;
    component = render(
      <Provider store={store}>
        <TablePagination gamesPages={gamesPages} pageIndex={0} setPageIndex={clickHandler} />
      </Provider>
    );
    container = component.container;
    asFragment = component.asFragment;
  });

  it("exists", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("has correct page count", () => {
    const pageCount = container.querySelectorAll("li").length;
    expect(pageCount).toBe(4);
  });

  it("selects a page", async () => {
    const button = getByText(container, "2");
    expect(hasClicked).toBe(false);
    expect(indexNumber).toBe(-1);
    fireEvent.click(button);
    expect(hasClicked).toBe(true);
    expect(indexNumber).toBe(1);
  });

});