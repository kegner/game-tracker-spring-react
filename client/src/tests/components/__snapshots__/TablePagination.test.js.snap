// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePagination exists 1`] = `
<DocumentFragment>
  <nav
    aria-label="Page navigation example"
  >
    <ul
      class="pagination my-1"
    >
      <li
        class="page-item"
      >
        <button
          class="page-link"
          disabled=""
        >
          Previous
        </button>
      </li>
      <li
        class="page-item  active"
      >
        <button
          class="page-link"
        >
          1
        </button>
      </li>
      <li
        class="page-item "
      >
        <button
          class="page-link"
        >
          2
        </button>
      </li>
      <li
        class="page-item"
      >
        <button
          class="page-link"
        >
          Next
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
